### PR TITLE
Add staging step to stage artifacts before signing

### PR DESCRIPTION
**Pull request by chidozieononiwu**
This Fixes a bug in the release stage.
Stage artifacts such that only packages present in `ci.yml` are signed.

### CODE REVIEW

File: eng/pipelines/templates/stages/archetype-net-release.yml [hunk 1]
```diff
@@ -23,14 +23,21 @@ stages:
                   artifact: ${{parameters.ArtifactName}}
                   timeoutInMinutes: 5
 
+                - ${{ each artifact in parameters.Artifacts }}:
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-net-release.yml:26]: I'd prefer we not do this here but instead do it in the build/package stage and use it to omit any packages that aren't in the ci.yml from being published to the artifacts.

**Comment by chidozieononiwu** [comment]
This is good to go

**Comment by pakrym** [comment]
Why do we end-up packing so many packages and need a separate step to filter them? I thought `dotnet pack eng/service.proj /p:ServiceDirectory=${{parameters.ServiceToBuild}}` takes care of that.

**Comment by chidozieononiwu** [comment]
> Why do we end-up packing so many packages and need a separate step to filter them? I thought `dotnet pack eng/service.proj /p:ServiceDirectory=${{parameters.ServiceToBuild}}` takes care of that.

${{parameters.ServiceToBuild}} == '*' in `sdk/core/ci.yml` which effectively includes all the .csproj projects in the sdk directory. 
https://github.com/Azure/azure-sdk-for-net/blob/ee4200957ef7b53e4502a87ce562339ea021e7fa/eng/service.proj#L17

**Comment by chidozieononiwu** [comment]
Test Issue Comment

**Comment by weshaggard** [comment]
@pakrym do you think it is worth only calling build on the other libraries and then only calling pack on the list of packages in the ci.yml? I kind of feel that would be better then this filtering process we are currently doing but I'd like to make sure there isn't any value in calling pack on the other libraries. Does pack add any extra level of validation that we'd like to keep in this scenario?

**Comment by pakrym** [comment]
Yes, I think it's worth having different parameters for testing and packing. 

There is some minor value to it around verifying repo-wide packing-related props/targets changes. But I expect we'll discover regressions in those quickly enough either way.

**Comment by weshaggard** [comment]
@chidozieononiwu can you change the pack step to just call build and then use the artifact list to call pack on just those packages.

**Comment by chidozieononiwu** [comment]
> > @chidozieononiwu can you change the pack step to just call build and then use the artifact list to call pack on just those packages.
> 
> What about having two pipeline parameters? One for serviceDirectory for tests (`*` in core) and one for pack (`core`)?

Is this instead of splitting the steps into build and pack?

**Comment by weshaggard** [comment]
@pakrym we could have ServiceDirectory set to core which would mean we would call pack on the core directory and then we could add another TestServiceDirectory option that would default to ServiceDirectory but we would override in the case of core and set it to "*".

**Comment by chidozieononiwu** [comment]
> @pakrym we could have ServiceDirectory set to core which would mean we would call pack on the core directory and then we could add another TestServiceDirectory option that would default to ServiceDirectory but we would override in the case of core and set it to "*".



> @pakrym we could have ServiceDirectory set to core which would mean we would call pack on the core directory and then we could add another TestServiceDirectory option that would default to ServiceDirectory but we would override in the case of core and set it to "*".

In that case we just have to do the Pack with ServiceDirectory which is already set to core. https://github.com/Azure/azure-sdk-for-net/blob/d7fe584345eb6742d4866bf9097ba1b784b6de4b/sdk/core/ci.yml#L35.
Right ServiceToBuild is only used in cases where it is present. Which happens to be only in core. https://github.com/Azure/azure-sdk-for-net/blob/d7fe584345eb6742d4866bf9097ba1b784b6de4b/eng/pipelines/templates/stages/archetype-sdk-client.yml#L36

**Comment by weshaggard** [comment]
@chidozieononiwu I didn't realize that we already had this. It might be worth renaming it to ServicesToTest to make it reflect it's new role.

**Review by weshaggard — APPROVED** [review]

**Pull request merged**